### ISSUE TITLE
Change fixed length character fields to variable-length strings

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#218]
Currently there are PM fields that have => 2000 characters (?). 

Right now we are truncating them (I believe the field lengths are defined in models.py). We should change those fields to the Postgres text field (I believe all the fields are in the postgress db) which is a variable-length character string that allows unlimited characters.

http://www.postgresql.org/docs/9.1/static/datatype-character.html

**Comment by RDmitchell** [comment]
@nicholasserra -- I am putting this in your queue, but we probably won't do anything about it until we work through our database restructuring thoughts, because the solution to this might change depending on what we decide to do with the database.

**Comment by nicholasserra** [comment]
This is going to be blocked by Django 1.9 as I don't want to make any more field changes before fixing Organization migrations.

**Comment by nicholasserra** [comment]
@RDmitchell Should we be using large text fields for all new fields for the refactor? cc @maronnax 

I'll investigate performance implications on this.

**Comment by nicholasserra** [comment]
Also, this ticket is now probably invalid.

**Comment by RDmitchell** [comment]
Yes, I'm going to close it.

I am a bit unclear about "large text fields" for all new fields for the refactor -- seems like they should have the field types they had in the building snapshot table, except that the some of the strings should be able to accommodate 4000+ characters -- there is an old issue for this, I will dredge it up and add the field names to one of the Sprint 2 issues

**Comment by nicholasserra** [comment]
The fields in the building snapshot table were too short, which is why this issue was created. So basically i'm asking if we should make all the fields support 4000+

**Comment by RDmitchell** [comment]
Looking at the character fields in the buildingsnapshot table. I think the only ones that need to hold 4000+ characters are those I listed in Issue #921, which are:

* property_notes
* energy_alerts
* space_alerts

And actually, looking at the definition for buildingsnapshot, all three of those are set to TextField, so they should remain that way. 

**So I think this might be a non-issue.**

The other two fields that can have 4000+ characters are in the Blob field, and listed in #921

**Comment by RDmitchell** [comment]
Would it make sense to make a unit test for this, ie, make sure that the long PM fields are not truncated upon import and when the data is actually put into the final db tables?

**Comment by nicholasserra** [comment]
Well all fields are truncated down to size when buildings are saved. If we removed that code we would immediately see when someone is trying to import something that is too long.

**Comment by RDmitchell** [comment]
What do you mean all fields are truncated down to size when buildings are saved? 

And we want to preserve the data in those 5 PM fields that are 4000+ characters, so what do you mean "import something that is too long"?

**Comment by nicholasserra** [comment]
I mean that when a building is saved, if the address is longer than 255 characters, it's trimmed down:

https://github.com/SEED-platform/seed/blob/develop/seed/models.py#L1350-L1402

All fields have a limit, we need to decide what that limit should be.

**Comment by RDmitchell** [comment]
I think we have to be careful not to truncate information that the user needs. I think a lot of the building snapshot fields defined as  CharField will probably almost always be < 255 (like postal_code and telephone, etc). But it is true that quite a few of them might be > 255, like address, property_name, building_certification, etc. So if it makes the most sense to make them all as TextField, then I suppose we could do that.
